controller/deploy_controller.rb

```diff
@@ -15,10 +18,13 @@ def initialize(environ, git_helper, user_comms, develop_tag_generator, other_tag
   def environment_choice
     @environ.downcase
     @tags_for_this_commit = @git_helper.get_tags_for_this_commit
+    dev_tag = CONFIG['develop']
+    test_tag = CONFIG['test']
+    stage_tag = CONFIG['stage']
```

Comment:
Much better (no string literals any more), but only part of the generalisation I had in mind. The actual stages of promotion are still hardcoded. You can leave it for now, but it would be a next step to have only a differentiation between first deployment step (dev) and further steps down the line (test, stage, preprod, prod...)